pkg/storage/database/database.go

```diff
@@ -26,6 +28,7 @@ type Database interface {
 	GetShareQuery(ctx context.Context, queryId uuid.UUID) (models.SharedQuery, bool)
 
 	GetUser(int64) *models.User
+	GetTeamId(userId uint) (uint, error)
```

Comment:
This is a nitpick we can change in a future PR: 

With our data model, a user can belong to multiple teams. So if you call `GetUser()`, then `models.User` also return a `[]Team`.

I'd suggest having the router be responsible for figuring out the team related to our request. Right now a user can only belong to one team, but in the future we'll add a level of indirection: `/dashboard/<team>/...` so we know how to manage requests.

Not a change that _needs_ to be made right now but since this is new code I wanted to point it out.